content/blog/nats-server-29-release.md

```diff
@@ -0,0 +1,489 @@
+
+For use cases that desire a large number of consumers, making them as lightweight as possible can help with scale and performance, such as one replica and having the state in memory.
+
+```go
+js.AddConsumer("EVENTS", &nats.ConsumerConfig{
+  Durable: "processor",
+  Replicas: 1,
+  MemoryStorage: true,
+})
+```
+
+For applications having consumers created prior to 2.8.3 or consumers created without those settings defined, these configuration options could not be changed.
+
+This release brings the ability to update both `Replicas` and `MemoryStorage` after consumer creation.
+
+In addition, the `FilterSubject` field is now able to be changed after creation.
+
+```go
+js.UpdateConsumer("EVENTS", &nats.ConsumerConfig{
+  Durable: "processor",
+  Replicas: 1,
+  MemoryStorage: true,
+  FilterSubject: "events.eu.*",
+})
+```
+
+#### Server tags and moving streams
+
+Server tags and placement of streams by tag were first introduced in 2.7.4. Each server in a cluster could be configured with zero or more tags indicating some property about them. For example, a cluster deployed in AWS within the us-east-2 region where each node is in a different availability zone.
+
+```
+// Node 1 configuration
+server_tags: ["cloud:aws", "region:us-east-2", "az:us-east-2a"]
+
+// Node 2 configuration
+server_tags: ["cloud:aws", "region:us-east-2", "az:us-east-2b"]
+
+// Node 3 configuration
+server_tags: ["cloud:aws", "region:us-east-2", "az:us-east-2c"]
+```
+
+A stream could then be defined with a *placement* declaration indicating the subset of nodes in a cluster that the stream replicas should exist on.
+
+```go
+js.AddStream(&nats.StreamConfig{
+  Name: "EVENTS",
+  Placement: &nats.Placement{
+    Tags: []string{"cloud:aws", "region:us-east-2"},
+  },
+})
+```
+
+When the server received this request it would determine whether there are a set of servers that have _all_ of these tags. If they have sufficient resources available to host a replica of the stream (assuming an R3 or R5), then that server would be elected.
+
+The `unique_tag` server config was later introduced in 2.8.0 to declare which tag (by prefix) must be _unique_ with respect to the replicas of a stream. Prior to this addition, it could be possible that all replicas would be placed on the same node (which would defeat the purpose of replication).
+
+```
+jetstream {
+  unique_tag: "az:"
+}
+```
+
+With a server having these additional configurations, placement of replicas for a stream would guarantee to be in different availability zones (being the most common use case).
+
+Stream placement becomes even more flexible by supporting **changing** the placement declaration in order to transparently move a stream to new servers. This can be done by simply doing `js.UpdateStream` with a new `Placement` configuration.
+
+## Operability and Security
```

Comment:
Maybe around this section mention that we are building with Go 1.19 so can use GOMEMLIMIT to apply cgroups limits?

Reply:
Ah, good call out. Yes that is definitely worth noting.

Reply:
There is the automaxprocs too to apply proper cpus limits too.

Reply:
Actually maybe that could go under the stability section with the framing of "more control/predictability of memory usage"

Reply:
yeah or maybe highlight a note about containerized deployments where this applies mostly

Reply:
@wallyqs Added the new section, let me know if it makes sense.

Reply:
Thanks! Added some more details

Reply:
Great additions, thanks!